src/vec.rs

```diff
@@ -246,12 +246,27 @@ macro_rules! impl_vecn_float_methods {
 
         /// Returns `self` normalized to length 1.0.
         ///
-        /// For valid results, `self` must _not_ be of length zero.
+        /// For valid results, `self` must _not_ be of length zero, nor very close to zero.
+        /// You can check the results with [`Self::is_finite`], or use [`Self::normalize_or_zero`].
         #[inline(always)]
         pub fn normalize(self) -> Self {
             Self($flttrait::normalize(self.0))
         }
 
+        /// Returns `self` normalized to length 1.0 if possible, else returns zero.
+        ///
+        /// In particular, if the input is zero (or very close to zero), or non-finite,
+        /// the result of this operation will be zero.
+        #[inline(always)]
+        pub fn normalize_or_zero(self) -> Self {
+            let rcp = self.length_recip();
+            if rcp.is_finite() {
+                self * rcp
```

Comment:
This works because `length_recip()` returns `NAN` for infinite vectors. Can we trust that behavior is cross platform? Does the CI tests both SIMD on and SIMD off?

Reply:
I would expect it's IEEE behaviour so it should be cross platform. I do test with and without simd enabled. Vec2 and Vec3 won't be using simd in any case.

Reply:
Oh wait, I found the problem. I was thinking of `INFINITY * Vec2::X` as `Vec2::new(INFINITY, 0.0)`, but the are not equivalent. I will fix my tests